src/Panel/Controller/Controller.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Kirby\Panel\Controller;
+
+use Kirby\Cms\App;
+use Kirby\Http\Request;
+
+/**
+ * @package   Kirby Panel
+ * @author    Nico Hoffmann <nico@getkirby.com>
+ * @link      https://getkirby.com
+ * @copyright Bastian Allgeier
+ * @license   https://getkirby.com/license
+ * @since     6.0.0
+ * @unstable
+ *
+ * @codeCoverageIgnore
+ */
+abstract class Controller
+{
+	protected App $kirby;
+	protected Request $request;
+
+	public function __construct()
+	{
+		$this->kirby   = App::instance();
+		$this->request = $this->kirby->request();
+	}
+
+	abstract public function load();
```

Comment:
What are actually the allowed return types here? Array? True? String for error?

Reply:
It depends on whatever we accept in our response handlers for each context.